File: CoreVia/CoreVia/Models/TrainerModels.swift
//
//  TrainerModels.swift
//  CoreVia
//

import Foundation
import SwiftUI

// MARK: - Trainer Response
struct TrainerResponse: Codable, Identifiable, Hashable {
    static func == (lhs: TrainerResponse, rhs: TrainerResponse) -> Bool { lhs.id == rhs.id }
    func hash(into hasher: inout Hasher) { hasher.combine(id) }

    let id: String
    let name: String
    let profileImageUrl: String?
    let specialization: String?
    let experience: Int?
    let rating: Double?
    let pricePerSession: Double?
    let bio: String?
    let verificationStatus: String
    let instagramHandle: String?

    enum CodingKeys: String, CodingKey {
        case id, name, specialization, experience, rating, bio
        case profileImageUrl = "profile_image_url"
        case pricePerSession = "price_per_session"
        case verificationStatus = "verification_status"
        case instagramHandle = "instagram_handle"
    }

    var category: TrainerCategory {
        guard let spec = specialization?.lowercased() else { return .fitness }
        if spec.contains("yoga") { return .yoga }
        if spec.contains("cardio") { return .cardio }
        if spec.contains("nutrition") || spec.contains("qidalanma") { return .nutrition }
        if spec.contains("strength") || spec.contains("guc") { return .strength }
        return .fitness
    }

    var specialtyTags: [TrainerCategory] {
        var tags: [TrainerCategory] = [category]
        guard let spec = specialization?.lowercased() else { return tags }
        for cat in TrainerCategory.allCases where cat != category {
            if spec.contains(cat.rawValue.lowercased()) {
                tags.append(cat)
            }
        }
        return tags
    }

    var displayRating: String {
        guard let r = rating else { return "--" }
        return String(format: "%.1f", r)
    }

    var displayPrice: String {
        guard let p = pricePerSession else { return "--" }
        return String(format: "%.0f ₼", p)
    }

    var displayExperience: String {
        let loc = LocalizationManager.shared
        guard let e = experience else { return "--" }
        return "\(e) \(loc.localized("trainer_years_short"))"
    }
}

// MARK: - Trainer Category
enum TrainerCategory: String, CaseIterable {
    case fitness = "Fitness"
    case strength = "Guc"
    case cardio = "Kardio"
    case yoga = "Yoga"
    case nutrition = "Qidalanma"

    var icon: String {
        switch self {
        case .fitness: return "figure.strengthtraining.traditional"
        case .strength: return "dumbbell.fill"
        case .cardio: return "heart.fill"
        case .yoga: return "figure.yoga"
        case .nutrition: return "leaf.fill"
        }
    }

    var color: SwiftUI.Color {
        switch self {
        case .fitness: return AppTheme.Colors.catFitness
        case .strength: return AppTheme.Colors.catStrength
        case .cardio: return AppTheme.Colors.catCardio
        case .yoga: return AppTheme.Colors.catYoga
        case .nutrition: return AppTheme.Colors.catNutrition
        }
    }

    var localizedName: String {
        let loc = LocalizationManager.shared
        switch self {
        case .fitness: return loc.localized("specialty_fitness")
        case .strength: return loc.localized("specialty_strength")
        case .cardio: return loc.localized("specialty_cardio")
        case .yoga: return loc.localized("specialty_yoga")
        case .nutrition: return loc.localized("specialty_nutrition")
        }
    }
}

File: CoreVia-Android/app/src/main/java/life/corevia/app/data/model/TrainerModels.kt
package life.corevia.app.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import life.corevia.app.ui.theme.*
import androidx.compose.ui.graphics.Color
import kotlin.math.abs

/**
 * iOS TrainerModels.swift equivalent
 * Trainer response, kateqoriya, UserResponse modelleri
 */

// ═══════════════════════════════════════════════════════════════════
// MARK: - Trainer Response
// ═══════════════════════════════════════════════════════════════════
@Serializable
data class TrainerResponse(
    val id: String = "",
    val name: String = "",
    @SerialName("profile_image_url") val profileImageUrl: String? = null,
    val specialization: String? = null,
    val experience: Int? = null,
    val rating: Double? = null,
    @SerialName("price_per_session") val pricePerSession: Double? = null,
    val bio: String? = null,
    @SerialName("verification_status") val verificationStatus: String = "",
    @SerialName("instagram_handle") val instagramHandle: String? = null
) {
    /** Auto-detect category from specialization */
    val category: TrainerCategory
        get() {
            val spec = specialization?.lowercase() ?: return TrainerCategory.FITNESS
            return when {
                spec.contains("yoga") -> TrainerCategory.YOGA
                spec.contains("cardio") || spec.contains("kardio") -> TrainerCategory.CARDIO
                spec.contains("nutrition") || spec.contains("qidalanma") -> TrainerCategory.NUTRITION
                spec.contains("strength") || spec.contains("guc") || spec.contains("güc") -> TrainerCategory.STRENGTH
                else -> TrainerCategory.FITNESS
            }
        }

    /** Get specialty tags from specialization text */
    val specialtyTags: List<TrainerCategory>
        get() {
            val tags = mutableListOf(category)
            val spec = specialization?.lowercase() ?: return tags
            TrainerCategory.entries.forEach { cat ->
                if (cat != category && spec.contains(cat.value.lowercase())) {
                    tags.add(cat)
                }
            }
            return tags
        }

    /** Full name alias */
    val fullName: String get() = name

    /** Detected category alias */
    val detectedCategory: TrainerCategory get() = category

    /** Initials from name */
    val initials: String
        get() {
            val parts = name.split(" ")
            return if (parts.size >= 2) {
                "${parts[0].take(1)}${parts[1].take(1)}".uppercase()
            } else {
                name.take(2).uppercase()
            }
        }

    /** Is verified trainer */
    val isVerified: Boolean
        get() = verificationStatus.lowercase() == "verified"

    /** Experience years alias */
    val experienceYears: Int? get() = experience

    val displayRating: String
        get() = rating?.let { "%.1f".format(it) } ?: "--"

    val displayPrice: String
        get() = pricePerSession?.let { "%.0f ₼".format(it) } ?: "--"

    val displayExperience: String
        get() = experience?.let { "$it il" } ?: "--"
}

// ═══════════════════════════════════════════════════════════════════
// MARK: - Trainer Category
// ═══════════════════════════════════════════════════════════════════
enum class TrainerCategory(val value: String, val displayName: String) {
    FITNESS("Fitness", "Fitness"),
    STRENGTH("Guc", "Güc"),
    CARDIO("Kardio", "Kardio"),
    YOGA("Yoga", "Yoga"),
    NUTRITION("Qidalanma", "Qidalanma");

    val color: Color
        get() = when (this) {
            FITNESS -> CatFitness
            STRENGTH -> CatStrength
            CARDIO -> CatCardio
            YOGA -> CatYoga
            NUTRITION -> CatNutrition
        }
}

// ═══════════════════════════════════════════════════════════════════
// MARK: - User Response (Assign/Unassign API)
// ═══════════════════════════════════════════════════════════════════
@Serializable
data class UserResponse(
    val id: String = "",
    val name: String = "",
    val email: String = "",
    @SerialName("profile_image_url") val profileImageUrl: String? = null,
    @SerialName("user_type") val userType: String = "client",
    @SerialName("trainer_id") val trainerId: String? = null,
    @SerialName("is_premium") val isPremium: Boolean = false,
    val weight: Double? = null,
    val height: Double? = null,
    val age: Int? = null,
    val goal: String? = null
) {
    /** Full name alias */
    val fullName: String get() = name

    /** Name-dən baş hərflər (avatar üçün) */
    val initials: String
        get() {
            val parts = name.split(" ")
            return if (parts.size >= 2) {
                "${parts[0].take(1)}${parts[1].take(1)}".uppercase()
            } else {
                name.take(2).uppercase()
            }
        }

    /** Avatar rəngi — name hash ilə */
    val avatarColor: Color
        get() {
            val index = abs(name.hashCode()) % AvatarPalette.size
            return AvatarPalette[index]
        }
}
